### DISCUSSION TITLE
Spindle ON/OFF custom actions from ABB rapid - error

### DISCUSSION THREAD

**Discussion by dariomarino01** [HALRobotics/Beta#126] [Q&A] [answered]
Hi everyone,
I'm struggling with the spindle ON/OFF custom actions.
I get an error message when trying to export - as if there is a syntax error but I copy-pasted the code from a routine which is already in the controller.
Here is the file in case you can have a look.
Thanks
[Cambridge_Cell.zip](https://github.com/HALRobotics/Beta/files/8728567/Cambridge_Cell.zip)

**Comment by sebandraos** [comment, marked as answer]
The issue is here:
![image](https://user-images.githubusercontent.com/6084391/169820222-4f6519e3-94ff-47cf-a440-3521b0b8f30b.png)
Custom actions are not designed for adding entire functions to your controller. If this PROC is already defined in a system module on the controller then you can simply call it in the _CustomAction_ with `Spindle_On_Tool_1;`. If however, this isn't already defined in your controller then you have two options - a) add it to a system module or b) recreate it using a series of `Actions` and combine them into a `Procedure`. Personally I would go for option a in this instance because it's a static function which is intrinsically linked to your cell.
